pylxd/client.py

```diff
@@ -482,7 +482,11 @@ def authenticate(self, secret):
         if self.trusted:
             return
         cert = open(self.api.session.cert[0]).read().encode("utf-8")
-        self.certificates.create(secret, cert)
+
+        if self.has_api_extension("explicit_trust_token"):
+            self.certificates.create(password="", cert_data=cert, secret=secret)
+        else:
+            self.certificates.create(password=secret, cert_data=cert)
```

Comment:
Once again I believe we are denying passwork auth for LTS users, here I would suggest introducing a flag on the `authenticate` function (and this is an opportunity to put my new code suggestion skills to use). I like this even more because here we can convey token authentication as being the default option.

```suggestion
def authenticate(self, secret, use_token_auth=true):
        if self.trusted:
            return
        cert = open(self.api.session.cert[0]).read().encode("utf-8")

        if self.has_api_extension("explicit_trust_token") and use_trust_token:
            self.certificates.create(password="", cert_data=cert, secret=secret)
        else:
            self.certificates.create(password=secret, cert_data=cert)
```

I couldn't select one of the lines for some reason but this should get the point across.
Please let me know if you agree with this approach.